cmd/gardener-apiserver/app/gardener_apiserver.go

```diff
@@ -375,9 +379,39 @@ func (o *Options) Run(ctx context.Context) error {
 		return err
 	}
 
+	if err := server.GenericAPIServer.AddPostStartHook("bootstrap-public-info", func(pctx genericapiserver.PostStartHookContext) error {
+		p := publicInfo{
+			Version:      version.Get().String(),
+			FeatureGates: fmt.Sprint(features.DefaultFeatureGate),
+		}
+
+		if len(o.ExtraOptions.WorkloadIdentityTokenIssuer) != 0 {
+			p.WorkloadIdentityIssuerURL = &o.ExtraOptions.WorkloadIdentityTokenIssuer
+		}
+
+		marshalledInfo, err := yaml.Marshal(p)
+		if err != nil {
+			return err
+		}
+
+		controllerManager, err := publicinfo.NewController(kubeClient, marshalledInfo)
+		if err != nil {
+			return err
+		}
+		return controllerManager.Start(pctx.Context, log)
+	}); err != nil {
+		return err
+	}
+
 	return server.GenericAPIServer.PrepareRun().Run(ctx.Done())
 }
 
+type publicInfo struct {
+	Version                   string  `json:"version" yaml:"version"`
+	WorkloadIdentityIssuerURL *string `json:"workloadIdentityIssuerURL,omitempty" yaml:"workloadIdentityIssuerURL,omitempty"` // TODO(vpnachev): Switch from omitempty(*string) to omitzero(string) once Gardener is build with go 1.24, ref: https://tip.golang.org/doc/go1.24
+	FeatureGates              string  `json:"featureGates" yaml:"featureGates"`
```

Comment:
I'm not sure whether it makes sense to publish the feature gates of GAPI here. What about the feature gates of all the other components (GCM, gardenlet, ...)?

Reply:
I think they should be published because sometimes user-facing gardener behavior is controlled by these feature gates, but as of now the users does not have a uniform way to see what feature gates are enabled/disabled. Gardener users have to rely on the gardener installation maintainers to document somewhere else the state of the feature gates. 

The configmap is structured in such a way that other components will be able to share their own configuration details, e.g. the gardener-scheduler can share its scheduling strategy. It is up to each component what exactly to share in this configmap, if the component find it useful to have the its feature gates shared - why not? I guess GOP feature gates will not be of interest of an end user, but gardenlet and GCM could be.

Reply:
I agree it's helpful, but I'd prefer to do it consistently for all (relevant) components. Since the main focus of this PR is publishing the issuer URL, I'm fine with removing the feature gate population for now. I'm also fine with adding it for the remaining Gardener components, but I can understand if this exceeds the scope.

Reply:
The feature gates and gardener version us just long hanging fruit, that why I added them. 
So, should I remove the feature gates for now?

Reply:
Maybe yes, perhaps you could open an issue so that we can better track and condense this follow-up effort. Thank you!

Reply:
The issue: https://github.com/gardener/gardener/issues/11340